src/megaapi_impl.cpp

```diff
@@ -882,6 +930,194 @@ MegaHandle MegaNodePrivate::getOwner() const
 
 const unsigned int MegaApiImpl::MAX_SESSION_LENGTH = 64;
 
+
+MegaBackgroundMediaUploadPrivate::MegaBackgroundMediaUploadPrivate(MegaApiImpl* capi)
+    : api(capi)
+{
+    // generate fresh random encryption key/CTR IV for this file
+    api->client->rng.genblock(filekey, sizeof filekey);
+}
+
+MegaBackgroundMediaUploadPrivate::MegaBackgroundMediaUploadPrivate(const string& serialised, MegaApiImpl* capi)
+    : api(capi)
+{
+    CacheableReader r(serialised);
+    string mediapropertiesstr;
+    unsigned char expansions[8];
+    string fileattrstring; // fileattrstring is not serialized
+    if (!r.unserializebinary(filekey, sizeof(filekey)) ||
+        !r.unserializechunkmacs(chunkmacs) ||
+        !r.unserializestring(mediapropertiesstr) ||
+        !r.unserializestring(url) ||
+        !r.unserializeexpansionflags(expansions, 0))
+    {
+        LOG_err << "MegaBackgroundMediaUploadPrivate unserialization failed at field " << r.fieldnum;
+    }
+    else
+    {
+        mediaproperties = MediaProperties(mediapropertiesstr);
+    }
+}
+
+std::string MegaBackgroundMediaUploadPrivate::serialize()
+{
+    std::string s;
+    CacheableWriter w(s);
+    w.serializebinary(filekey, sizeof(filekey));
+    w.serializechunkmacs(chunkmacs);
+    w.serializestring(mediaproperties.serialize());
+    w.serializestring(url);
+    w.serializeexpansionflags(false, false);
+    return s;
+}
+
+
+MegaBackgroundMediaUploadPrivate::~MegaBackgroundMediaUploadPrivate()
+{
+}
+
+class MEGA_API EncryptFilePieceByChunks : public EncryptByChunks
+{
+    // specialisation for encrypting a piece of a file without using too much RAM
+    FileAccess* fain;
+    FileAccess* faout;
+    m_off_t inpos, outpos;
+    string buffer;
+    unsigned lastsize;
+
+public:
+
+    EncryptFilePieceByChunks(FileAccess* f1, m_off_t f1pos, FileAccess* f2, m_off_t f2pos, SymmCipher* k, chunkmac_map* m, uint64_t c)
```

Comment:
Despite there's a lot of legacy code not following this rules, we try to name variable members with a prefix `m`, and `c` for params in the ctor, specially in MEGAchat and in the new objects of the intermediate layer in SDK. I'll fix it